popupmenu for files

chrisnojima opened the pull request:
@keybase/react-hackers files menu has 'danger' items so i enhanced popup menu. also supports scrolling

chrisnojima commented on shared/common-adapters/popup-menu.desktop.js:
```diff
@@ -11,24 +11,47 @@ class Menu extends Component<void, Props, void> {
 
     return (
-      <Box style={{...stylesMenuCatcher, ...this.props.style}} onClick={() => this.props.onHidden()}>
+      <Box style={{...stylesMenuCatcher}} onClick={e => {
```
props.style don't actually belong here. this needs to cover the entire screen basically so it's margins shouldn't be messed with

chrisnojima commented on shared/folders/files/render.js.flow:
```diff
@@ -25,7 +25,7 @@ export type UnlockDevice = {
   theme: 'public' | 'private',
   visiblePopupMenu: boolean,
-  popupMenuItems: Array<MenuItem>,
+  popupMenuItems: Array<MenuItem | null>,
```
needed by flow for some reason. seems like a stricter subset should just work...

chromakode commented on shared/common-adapters/popup-menu.desktop.js:
```diff
@@ -11,24 +11,47 @@ class Menu extends Component<void, Props, void> {
+    .menu-hover-danger:hover .title { color: ${globalColors.white}; }
+    .menu-hover-danger .subtitle { color: ${globalColors.black_40}; }
+    .menu-hover-danger:hover .subtitle { color: ${globalColors.white}; }
     }
```
Is this an extra `}` at the end?

chromakode commented on shared/common-adapters/popup-menu.desktop.js:
```diff
@@ -11,24 +11,47 @@ class Menu extends Component<void, Props, void> {
-          ))}
+        <Box style={{...stylesMenu, ...this.props.style}}>
+          <Box style={{...globalStyles.flexBoxColumn, flexShrink: 0, paddingTop: 7, paddingBottom: 7}} >
+            {this.props.items.map((i, idx) => (i ? (
```
I think the structure of the `?` operator here is a bit hard to read since you have to hunt for the `:` at the end. Perhaps assign `separator = <Box ...>` and flip the order so it reads: `!i ? separator : ...` so the logic clearer?

chromakode commented:
:+1: % extra `}`

MarcoPolo commented on shared/common-adapters/popup-menu.js.flow:
```diff
@@ -4,12 +4,14 @@ import React, {Component} from 'react'
 
 export type Props = {
-  items: Array<MenuItem>,
+  items: Array<MenuItem | null>, // null for dividers
```
:sheep: a little weird

MarcoPolo commented:
👍

chrisnojima commented:
@keybase/react-hackers made dividers more specific and cleaned up a bunch of key warnings from the root branches. also fixed up that autocomplete warning which has been in there too long

chrisnojima commented:
@keybase/react-hackers ok this now includes fixes for all the parent branches. when i get another thumb i'm going to merge this all down

MarcoPolo commented:
👍
